Lib/pdb.py

```diff
@@ -1719,10 +1753,24 @@ def _help_message_from_doc(self, doc):
             elif i < usage_end:
                 prefix = "       "
             else:
+                if usage_only:
+                    break
                 prefix = ""
             formatted.append(indent + prefix + line)
         return "\n".join(formatted)
 
+    def _print_invalid_arg(self, arg):
+        """Return the usage string for a function."""
+
+        self.error(f"Invalid argument: {arg}")
+
+        # Yes it's a bit hacky. Get the caller name, get the method based on
+        # that name, and get the docstring from that method.
+        # This should NOT fail if the caller is a method of this class.
+        doc = inspect.getdoc(getattr(self, sys._getframe().f_back.f_code.co_name))
```

Comment:
To avoid this,  the `__doc__` of the caller could be passed in explicitly. 
This would add some verbosity to the calls, but these are already identical in terms of the arguments.

Reply:
There's no clean way to pass the ``__doc__`` of the caller inside the caller. I'd rather keep the ugliness in a single place and document it, than pollute everywhere. With this hack(which is not hard to understand as long as you are familiar with the frame/code structure), all the callers are very clean. Similar methods are actually used to get the docs for all the commands.

Reply:
I understand the motivation here. Both solutions are not ideal and if equally "unclean" pulling it in a single place does make sense. Whether this is the case here, may be just personal opinion....